x-pack/plugin/repositories-stats/src/main/java/org/elasticsearch/xpack/repositories/stats/action/TransportRepositoriesStatsAction.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.repositories.stats.action;
+
+import org.elasticsearch.action.FailedNodeException;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.nodes.TransportNodesAction;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.repositories.RepositoriesService;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportRequest;
+import org.elasticsearch.transport.TransportService;
+
+import java.io.IOException;
+import java.util.List;
+
+public final class TransportRepositoriesStatsAction extends TransportNodesAction<
```

Comment:
Should we remove `org.elasticsearch.xpack.searchablesnapshots.rest.RestRepositoryStatsAction` and associated logic now that we have this in this PR?

Reply:
We can remove it in a separate PR if we think it won't be useful anymore, wdyt?

Reply:
Fine as well :) just figured we shouldn't forget it.